### PR TITLE
[ReadMe]: Enabled generation script to add Bicep + JSON deployment examples to readme V2

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

Enabled generation script to add Bicep + JSON deployment examples to readme -> Based on parameter files (1 each)

# Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

### CODE REVIEW

File: utilities/tools/Set-ModuleReadMe.ps1 [hunk 1]
```diff
@@ -342,6 +342,159 @@ function Set-OutputsSection {
...
+                '```',
+                '',
+                '</details>'
+            )
+        }
+
+        if ($addBicep) {
+            $JSONParameters = (ConvertFrom-Json $contentInJSONFormat -AsHashtable -Depth 99).parameters
+
+            ## Handle KeyVaut references
```
Comment (MrMCake) [inline utilities/tools/Set-ModuleReadMe.ps1:398]:
```suggestion
            # Handle KeyVaut references
```

File: utilities/tools/Set-ModuleReadMe.ps1 [hunk 2]
```diff
@@ -299,6 +299,159 @@ function Set-OutputsSection {
...
+            }
+
+            $extendedKeyVaultReferences = @()
+            $counter = 0
+            foreach ($reference in ($keyVaultReferenceData | Sort-Object -Property 'vaultName' -Unique)) {
+                $counter++
+                $extendedKeyVaultReferences += @(
+                    "resource kv$counter 'Microsoft.KeyVault/vaults@2019-09-01' existing = {",
+                    ("    name: '{0}'" -f $reference.vaultName),
+                    ("    scope: resourceGroup('{0}','{1}')" -f $reference.subscriptionId, $reference.resourceGroupName),
```
Comment (MariusStorhaug) [inline utilities/tools/Set-ModuleReadMe.ps1:379]: Assuming we are going with 2*space indents.

```suggestion
                    ("  name: '{0}'" -f $reference.vaultName),
                    ("  scope: resourceGroup('{0}','{1}')" -f $reference.subscriptionId, $reference.resourceGroupName),
```

File: arm/Microsoft.AnalysisServices/servers/readme.md [hunk 3]
```diff
@@ -96,3 +154,245 @@ Create a role assignment for the given resource. If you want to assign a service
...
+<details>
+
+<summary>via Bicep module</summary>
+
+```bicep
+module servers './Microsoft.AnalysisServices/servers/deploy.bicep' = {
+  name: '${uniqueString(deployment().name)}-servers'
+  params: {
+      skuCapacity: 1
+      diagnosticWorkspaceId: '/subscriptions/<<subscriptionId>>/resourcegroups/validation-rg/providers/microsoft.operationalinsights/workspaces/adp-<<namePrefix>>-az-law-x-001'
```
Comment (MariusStorhaug) [inline arm/Microsoft.AnalysisServices/servers/readme.md:247]: Again assuming the examples are based on 2*space indentations. The script should kater for adding 4, not 6 spaces here.

```suggestion
module servers './Microsoft.AnalysisServices/servers/deploy.bicep' = {
  name: '${uniqueString(deployment().name)}-servers'
  params: {
    skuCapacity: 1
    diagnosticWorkspaceId: '/subscriptions/<<subscriptionId>>/resourcegroups/validation-rg/providers/microsoft.operationalinsights/workspaces/adp-<<namePrefix>>-az-law-x-001'
```

**Review by MariusStorhaug — CHANGES_REQUESTED** [review]
First check.

File: utilities/tools/Set-ModuleReadMe.ps1 [hunk 4]
```diff
@@ -299,6 +299,159 @@ function Set-OutputsSection {
     return $updatedFileContent
 }
```
[the remark this answers is not in the document]
Comment (MrMCake) [reply utilities/tools/Set-ModuleReadMe.ps1:301]: added

**Review by MariusStorhaug — DISMISSED** [review]

**Review by MariusStorhaug — APPROVED** [review]

**Pull request merged**